comm/src/main/scala/coop/rchain/comm/comm.scala

```diff
@@ -40,7 +43,7 @@ case class PeerNode(id: NodeIdentifier, endpoint: Endpoint) {
     s"rnode://$sKey@${endpoint.host}:${endpoint.udpPort}"
 
   def withUdpSocket(udp: java.net.InetSocketAddress): PeerNode =
-    PeerNode(id, endpoint.withUdp(udp))
+    new PeerNode(id, endpoint.withUdp(udp))
```

Comment:
Wait . . . what?

Reply:
yeah, this should be reverted back, forgot about it, let me elaborate on this a bit

`PeerNode` is case class but really should not be as you have some class other class that extends it. This breaks the equals method and you will have some unexpected errors on runtime if you leave it as it is. 
I tried to make `PeerNode` not a case class but then realized that I don't know enough of domain to touch it just yet. So I reverted that back (making `PeerNode` case class one more time) leaving a `TODO/FIX-ME` and a comment.

This is something I hope we could discuss really quickly and one of the reasons I like static analysis like that as it finds those kinds of errror - thus WartRemover

Regardless of `PeerNode` being case class or not, for now the above should be reverted back to previous version

Reply:
Got it; thanks.